### PR TITLE
Add "Get total expenditure" command

### LINKED ISSUE #55: As a tech-savvy NUS student, I can check my total expenditure by month and filter by category
...so that I can check that I am within my budget.

### PR DESCRIPTION
**Pull request by woojiahao**
Closes #55

### CODE REVIEW

**Review by sp4ce-cowboy — CHANGES_REQUESTED** [review]
Motly ok, just some minor nits, see comment

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -18,6 +18,7 @@ public class Messages {
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
     public static final String MESSAGE_DUPLICATE_FIELDS =
                 "Multiple values specified for the following single-valued field(s): ";
+    public static final String MESSAGE_INVALID_MONTH = "Month must be between 1 and 12 (inclusive).";
```
[the remark this answers is not in the document]
Comment (woojiahao) [reply src/main/java/seedu/address/logic/Messages.java:21]: Oh oops, the duplication is abit confusing, ok

File: src/main/java/seedu/address/logic/commands/transaction/GetTotalExpenditureCommand.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (month < 1 || month > 12) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MONTH);
+        }
+
+        model.updateFilteredTransactionList(transaction -> {
+            boolean isExpense = transaction.getType().type.equals(TransactionType.EXPENSE);
+            boolean isSameMonth = transaction.getDateTime().dateTime.getMonthValue() == month;
+            if (categoryFilter == null || categoryFilter.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (woojiahao) [reply src/main/java/seedu/address/logic/commands/transaction/GetTotalExpenditureCommand.java:52]: It's covered, the `categoryFilter.isEmpty()` call checks if the category is empty. There's no way to specify an "empty" category so just `get_total_expenditure 5` works

File: src/main/java/seedu/address/logic/Messages.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (woojiahao) [reply src/main/java/seedu/address/logic/Messages.java:21]: Fixed

File: src/main/java/seedu/address/logic/commands/transaction/GetTotalExpenditureCommand.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sp4ce-cowboy) [reply src/main/java/seedu/address/logic/commands/transaction/GetTotalExpenditureCommand.java:52]: Ah I see okok

**Review by sp4ce-cowboy — APPROVED** [review]
LGTM!

**Pull request merged**